src/client.rs

```diff
@@ -138,11 +183,26 @@ pub mod future {
     pub struct ConnectFuture<Req, Resp, E>
         where Req: Serialize + 'static,
               Resp: Deserialize + 'static,
-              E: Deserialize + 'static,
+              E: Deserialize + 'static
     {
+        #[cfg(not(feature = "tls"))]
+        inner:
+            future::Either<
```

Comment:
Can we add a type alias for part of this so we don't have to rewrite the entire thing in both branches of the `cfg`?